sdk/contentsafety/azure-ai-contentsafety/src/main/java/com/azure/ai/contentsafety/ContentSafetyClientBuilder.java

```diff
@@ -283,8 +284,17 @@ private ContentSafetyClientImpl buildInnerClient() {
         return client;
     }
 
+    @Generated
+    private void validateBuilderParameters() {
+        Objects.requireNonNull(endpoint, "'endpoint' cannot be null.");
+
+        // dev may customize and add more validation here
+        // in this module, maybe check either keyCredential or tokenCredential exists
+    }
+
     @Generated
     private HttpPipeline createHttpPipeline() {
+        validateBuilderParameters();
```

Comment:
Why here and not in `buildInnerClient`? It doesn't matter much in this example; are there others where it might matter more?

Reply:
An alternative code flow would be user setting the `pipeline` via 
https://github.com/Azure/azure-sdk-for-java/blob/020a2a694b806c9640ee7583d9b578832ab71756/sdk/contentsafety/azure-ai-contentsafety/src/main/java/com/azure/ai/contentsafety/ContentSafetyClientBuilder.java#L88

I think in that branch, there is no need to verify anything (as user provides a pipeline and we cannot inspect it anyway).